src/java.net.http/share/classes/java/net/http/HttpClient.java

```diff
@@ -354,16 +355,59 @@ public interface Builder {
          */
         public Builder authenticator(Authenticator authenticator);
 
+        /**
+         * Binds the socket to this local address when creating
+         * connections for sending requests.
+         *
+         * <p> If no local address is set or {@code null} is passed
+         * to this method then sockets created by the
+         * HTTP client will be bound to an automatically
+         * assigned socket address.
+         *
+         * <p> Common usages of the {@code HttpClient} do not require
+         * this method to be called. Setting a local address, through this
+         * method, is only for advanced usages where users of the {@code HttpClient}
+         * require specific control on which network interface gets used
+         * for the HTTP communication. Callers of this method are expected to
+         * be aware of the networking configurations of the system where the
+         * {@code HttpClient} will be used and care should be taken to ensure the
+         * correct {@code localAddr} is passed. Failure to do so can result in
+         * requests sent through the {@code HttpClient} to fail.
+         *
+         * @implSpec The default implementation of this method throws
+         * {@code UnsupportedOperationException}. {@code Builder}s obtained
```

Comment:
I think the implSpec here is correct, but will be confusing for most users. I'm not sure what value it adds. Do we really need it?

Reply:
Hello Michael,
Most users will be using the `HttpClient.newBuilder()` to create the builder, so this note about `UnsupportedOperationException` can be confusing. However, for implementations (libraries?) which provide their  own implementation of the `java.net.http.HttpClient.Builder`, I think, this note would be relevant. This approach is similar to what we already have on `java.net.http.HttpClient.newWebSocketBuilder()` method.

Reply:
Sure, I just think when most developers read "The default implementation of this method throws UOE" they will think they can't use it without implementing something themselves. Library developers will figure it out anyway.

Reply:
This is the standard wording that has been used throughout the JDK to document the behavior of default methods on interfaces. Unless we received different feedback during the CSR review I'd suggest to leave it that way. The second sentence makes it clear that our concrete implementations override that default behavior.